### DISCUSSION TITLE
Does it allow to send HTTP request in the callback function of batch API

### DISCUSSION THREAD

**Discussion by mrdulin** [reduxjs/react-redux#2203] [Q&A]
I saw code like below, the code does not throw error at runtime. But I don't think put I/O operations in the callback function of  `batch` is correct. However, I am not aware of what effects it might have. If it's a bad practice, please document this in the instructions.

```js
batch(async () => {
  const res = await request(/***/);
  dispatch(action1(res));
  dispatch(action2(res));
})
```

**Comment by markerikson** [comment]
That callback is really meant for dispatching synchronous actions so that there's only one UI update.

That said, it's also obsolete as of React 18, as React 18 now _always_ batches all queued renders within a single event loop tick.  So, if you're using React 18, you don't need to use `batch` at all.
